### PR TITLE
Add Instance Methods ptp to NumPy Frontend

### PR DESCRIPTION
**Pull request by indrawatideasy**
Subtask https://github.com/unifyai/ivy/issues/13611 for Instance Methods to NumPy Frontend https://github.com/unifyai/ivy/issues/3607

### CODE REVIEW

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 1]
```diff
@@ -471,3 +471,13 @@ def __setitem__(self, key, value):
 
     def __mod__(self, value, /):
         return np_frontend.mod(self._ivy_array, value, out=self)
+    
+    def ptp(self, *, axis=None, dtype=None, out=None, keepdims=False, where=True):
+        return np_frontend.ptp(
```
Comment (hmahmood24) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:476]: There is no method `ptp` present in the `np_frontend` namespace. Did you try running any tests locally to confirm that they pass? The [CI tests](https://github.com/unifyai/ivy/actions/runs/4632978419/jobs/8197683085?pr=13696#step:3:123) fail essentially because of this. If there is no function `ptp` in the `np_frontend` or `ivy` namespace, then you will have to implement the logic of your function as a composition of other `ivy` functions such as `ivy.max`, `ivy.min` etc.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 2]
```diff
@@ -2722,3 +2722,42 @@ def test_numpy_instance_mod__(
...
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="numpy.array",
+    method_name="ptp",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("float"),
+        min_axis=-1,
+        max_axis=0,
+        min_num_dims=1,
+        force_int_axis=True,
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2735]: We need to test the function with `tuple` of ints in the `axis` parameter as well since the [original method](https://numpy.org/doc/stable/reference/generated/numpy.ptp.html) supports that.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 3]
```diff
@@ -2722,3 +2722,42 @@ def test_numpy_instance_mod__(
...
+    
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="numpy.array",
+    method_name="ptp",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("float"),
+        min_axis=-1,
+        max_axis=0,
+        min_num_dims=1,
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2734]: Please don't restrict the `min_num_dims` to 1. We need to keep our tests coverage maximum and so test the function with arbitrary dimensions

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 4]
```diff
@@ -471,3 +471,13 @@ def __setitem__(self, key, value):
 
     def __mod__(self, value, /):
         return np_frontend.mod(self._ivy_array, value, out=self)
+    
+    def ptp(self, *, axis=None, dtype=None, out=None, keepdims=False, where=True):
```
Comment (hmahmood24) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:475]: The [original method](https://numpy.org/doc/stable/reference/generated/numpy.ptp.html) does not take `dtype` as an argument. Hence, it's fine to remote it from here.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 5]
```diff
@@ -2722,3 +2722,42 @@ def test_numpy_instance_mod__(
...
+    input_dtypes, x, axis = dtype_x_axis
+    helpers.test_frontend_method(
+        init_input_dtypes=input_dtypes,
+        method_input_dtypes=input_dtypes,
+        init_all_as_kwargs_np={
+            "object": x[0],
+        },
+        method_all_as_kwargs_np={
+            "axis": axis,
+            "dtype": "float64",
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2755]: The `dtype` argument can be removed since the `method_input_dtypes` argument of our test helper takes care of that already.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 6]
```diff
@@ -2722,3 +2722,42 @@ def test_numpy_instance_mod__(
...
+            "out": None,
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2756]: This can also be removed. The `out` parameter is already handled by our testing helpers

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 7]
```diff
@@ -2722,3 +2722,42 @@ def test_numpy_instance_mod__(
...
+    method_name="ptp",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("float"),
+        min_axis=-1,
+        max_axis=0,
+        min_num_dims=1,
+        force_int_axis=True,
+    ),
+)
+def test_numpy_ndarray_ptp(
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2738]: The name of the test function should be `test_numpy_instance_ptp`

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @indrawatideasy, thanks a lot for the PR. I have made some comments and suggested changes for you to address. I have to mention some points though:

1. There are some linting and formatting checks that your PR is failing. Kindly look into those [here](https://github.com/unifyai/ivy/actions/runs/4632978425/jobs/8197682858?pr=13696#step:5:126) and fix them. I'm guessing you haven't installed `pre-commit` since it automatically runs all these checks whenever you commit your code and push it. I suggest you [set up](https://lets-unify.ai/ivy/overview/contributing/setting_up.html#pre-commit) pre-commit in your local environment and [run](https://lets-unify.ai/ivy/overview/deep_dive/formatting.html#pre-commit-hook) these linting and formatting checks because otherwise the CI tests will fail every time you submit a PR that has linting and formatting issues.
2. I'm guessing you didn't run the tests locally because `pytest` would have thrown an error since there is no method `ptp` in the `np_frontend` namespace. You can check the failing CI tests [here](https://github.com/unifyai/ivy/actions/runs/4632978419/jobs/8197683085?pr=13696#step:3:123) to verify. Kindly always make sure that you run all your tests locally and ensure that they pass for ALL backends because otherwise the CI tests will fail when you submit a PR.

When you've done all these changes and verified that the tests are passing, feel free to ask for a re-review from me 🙂

Good luck!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.
